spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -473,12 +473,22 @@ def foo; :bar; end
       expect { raise FooError }.to raise_error(an_error_with_attribute(:foo).equal_to(:bar))
     end
 
+    it 'passes when the matcher matches the exception message' do
+      expect { raise FooError, "food" }.to raise_error(a_string_including("foo"))
+    end
+
     it 'fails with a clear message when the matcher does not match the raised error' do
       expect {
         expect { raise FooError }.to raise_error(an_error_with_attribute(:foo).equal_to(3))
       }.to fail_including("expected an error with attribute :foo equal to 3, got #<FooError: FooError>")
     end
 
+    it 'fails with a clear message when the matcher does not match the exception message' do
+      expect {
+        expect { raise FooError, "food" }.to raise_error(a_string_including("bar"))
+      }.to fail_including('expected a string including "bar", got #<FooError: food')
```

Comment:
We're not expecting a string including "bar" -- we're expecting an exception with a message that matches that.  It would be nice if the failure message said `expected Exception with message matching (a string including "bar")` (or something like that -- basically whatever `raise_error(Exception, a_string_including("bar"))` fails with now).  It looks like there's another special case for `Regexp` for that kind of phrasing in place that we should expand to work for matchers, too.